assets/js/stripe.js

```diff
@@ -398,305 +508,281 @@ jQuery( function( $ ) {
+			/**
+			 * ToDo: The logic here was restructured a bit, make sure it still works well.
+			 */
 
-				wc_stripe_form.block();
+			// If a source, token, or an intent is already in place, submit the form as usual.
+			if ( wc_stripe_form.isStripeSaveCardChosen() || wc_stripe_form.hasSource() || wc_stripe_form.hasToken() || wc_stripe_form.hasIntent() ) {
+				return true;
+			}
 
-				// Stripe Checkout.
-				if ( 'yes' === wc_stripe_params.is_stripe_checkout && wc_stripe_form.isStripeModalNeeded() && wc_stripe_form.isStripeCardChosen() ) {
-					if ( 'yes' === wc_stripe_params.is_checkout ) {
-						return true;
-					} else {
-						wc_stripe_form.openModal();
-						return false;
-					}
+			// Open the Stripe Checkout modal.
+			if ( 'yes' === wc_stripe_params.is_stripe_checkout && wc_stripe_form.isStripeModalNeeded() && wc_stripe_form.isStripeCardChosen() ) {
+				if ( 'yes' === wc_stripe_params.is_checkout ) {
+					return true;
+				} else {
+					wc_stripe_form.openModal();
+					return false;
 				}
+			}
 
-				/*
-				 * For methods that needs redirect, we will create the
-				 * source server side so we can obtain the order ID.
-				 */
-				if (
-					wc_stripe_form.isBancontactChosen() ||
-					wc_stripe_form.isGiropayChosen() ||
-					wc_stripe_form.isIdealChosen() ||
-					wc_stripe_form.isAlipayChosen() ||
-					wc_stripe_form.isSofortChosen() ||
-					wc_stripe_form.isP24Chosen() ||
-					wc_stripe_form.isEpsChosen() ||
-					wc_stripe_form.isMultibancoChosen()
-				) {
-					if ( $( 'form#order_review' ).length ) {
-						$( 'form#order_review' )
-							.off(
-								'submit',
-								this.onSubmit
-							);
-
-						wc_stripe_form.form.submit();
-
-						return false;
-					}
-
-					if ( $( 'form.woocommerce-checkout' ).length ) {
-						return true;
-					}
-
-					if ( $( 'form#add_payment_method' ).length ) {
-						$( 'form#add_payment_method' )
-							.off(
-								'submit',
-								this.onSubmit
-							);
+			// For methods that needs redirect, we will create the source server side so we can obtain the order ID.
+			if (
+				wc_stripe_form.isBancontactChosen() ||
+				wc_stripe_form.isGiropayChosen() ||
+				wc_stripe_form.isIdealChosen() ||
+				wc_stripe_form.isAlipayChosen() ||
+				wc_stripe_form.isSofortChosen() ||
+				wc_stripe_form.isP24Chosen() ||
+				wc_stripe_form.isEpsChosen() ||
+				wc_stripe_form.isMultibancoChosen()
+			) {
+				if ( $( 'form#order_review' ).length ) {
+					$( 'form#order_review' )
+						.off(
+							'submit',
+							this.onSubmit
+						);
 
-						wc_stripe_form.form.submit();
+					wc_stripe_form.form.submit();
 
-						return false;
-					}
+					return false;
 				}
 
-				wc_stripe_form.createSource();
+				if ( $( 'form#add_payment_method' ).length ) {
+					$( 'form#add_payment_method' )
+						.off(
+							'submit',
+							this.onSubmit
+						);
 
-				// Prevent form submitting
-				return false;
-			} else if ( $( 'form#add_payment_method' ).length ) {
```

Comment:
I don't think the refactored code handles this case.

Reply:
Yes, because we are not creating payment intents on the add payment method form anymore. The replacement is in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/796/files#diff-1f8d151e9960b8b51bc43b4a2aba09d4R504 , although I have no idea how I managed to mix the branches up for this 🤣